src/components/DropMenu/DropMenu.jsx

```diff
@@ -465,26 +482,39 @@ const DropMenu = createClass({
 					</ContextMenu.Target>
 					<ContextMenu.FlyOut className={cx('&', className)} style={flyOutStyle}>
 						{
-							_.map(nullOptions, (optionProps) => this.renderOption(optionProps, null))
-							.concat(_.isEmpty(nullOptions) ? [] : [(<div key={'OptionGroup-divider-NullOption'} className={cx('&-OptionGroup-divider')} />)])
-						}
-						{
-							joinArray(
-								// for each option group,
-								_.map(optionGroups, (optionGroupProps, optionGroupIndex) => {
-									const labelElements = rejectTypes(optionGroupProps.children, [DropMenu.Control, DropMenu.OptionGroup, DropMenu.Option, DropMenu.NullOption]);
-									// render label if there is one
-									return (_.isEmpty(labelElements) ? [] : [
-										<div {...optionGroupProps} className={cx('&-label', optionGroupProps.className)}>
-											{labelElements}
-										</div>,
-									// render the options in the group
-									]).concat(_.map(_.get(optionGroupDataLookup, optionGroupIndex), ({ optionProps, optionIndex }) => this.renderOption(optionProps, optionIndex, true)));
-								// append all ungrouped options as another unlabeled group
-								}).concat(_.isEmpty(ungroupedOptionData) ? [] : [_.map(ungroupedOptionData, ({ optionProps, optionIndex }) => this.renderOption(optionProps, optionIndex))]),
-								(element, index) => (<div key={`OptionGroup-divider-${index}`} className={cx('&-OptionGroup-divider')} />) // separate each group with divider
-							)
+							!_.isEmpty(headerProps) &&
+							<div
+								{...headerProps}
+								className={cx('&-Header', headerProps.className)}
+								onKeyDown={this.handleKeydown}
+							/>
 						}
+						<div
+							className={cx('&-option-container')}
+							style={_.assign({}, flyOutStyle, optionContainerStyle)}
```

Comment:
I needed to do this to maintain compatibility with the existing functionality; otherwise, even if the flyOutStyle's maxHeight is set to a smaller value, the optionContainerStyle will ignore it.